airbyte-json-validation/src/main/java/io/airbyte/validation/json/JsonSchemaValidator.java

```diff
@@ -25,13 +30,37 @@
 public class JsonSchemaValidator {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(JsonSchemaValidator.class);
+  // This URI just needs to point at any path in the same directory as /app/WellKnownTypes.json
+  // It's required for the JsonSchema#validate method to resolve $ref correctly.
+  private static final URI DEFAULT_BASE_URI;
+
+  static {
+    try {
+      DEFAULT_BASE_URI = new URI("file:///app/nonexistent_file.json");
```

Comment:
Why not use `file:///app/WellKnownTypes.json` instead?

Reply:
just to make it clear that it doesn't need to be the exact file being referenced. E.g. if we later added a second file, we wouldn't need to update this URI.

Reply:
Yes, the part that always feels a bit unusual is referencing a non existing file.

Reply:
hm. what if we appended the `nonexistent_file.json` in the constructor? so callers would just need `new JsonSchemaValidator(new URI("file:///app/"))`

Reply:
bump @gosusnp

Reply:
Not sure that appending this in the constructor helps understanding that much. It probably boils down to why do we actually need to pass a full URI rather than just the actual base URI.
I'd say that it doesn't feel ideal, at the same time, it doesn't seem important enough to block this PR. The comments helps with the understanding.